iphone/Classes/TiUINavBarButton.m

```diff
@@ -124,6 +124,9 @@ - (id)initWithProxy:(TiUIButtonProxy *)proxy_
   proxy = proxy_; // Don't retain
 
   self.accessibilityLabel = [proxy_ valueForUndefinedKey:@"accessibilityLabel"];
+  self.accessibilityValue = [proxy_ valueForUndefinedKey:@"accessibilityValue"];
+  self.accessibilityHint = [proxy_ valueForUndefinedKey:@"accessibilityHint"];
+  self.accessibilityIdentifier = [TiUtils composeAccessibilityIdentifier:self];
```

Comment:
TiUINavBarButton is neither derived from TiUIView nor UIAccessibilityElement. So accessibilityIdentifier will always be nil as per logic written in function 'composeAccessibilityIdentifier'. We can not make it nil.